packages/synapse-react-client/src/components/doi/CreateOrUpdateDoiModal.tsx

```diff
@@ -0,0 +1,324 @@
+    },
+  )
+
+  const entityType = entity
+    ? convertToEntityType(entity.concreteType)
+    : EntityType.FILE
+  const entityTypeDisplay = entityTypeToFriendlyName(entityType)
+  let versionHelpMarkdown = `A DOI can be associated with a specific version of this ${entityTypeDisplay}.
+  
+  Versioned DOIs will link to the specified version of the ${entityTypeDisplay}.
+  
+  A DOI without a version will always link to the newest version of this ${entityTypeDisplay},
+     so the data that someone retrieves using the DOI may change over time.`
+  if (entity && isTable(entity)) {
+    versionHelpMarkdown += `\n\nTo create a DOI that will always link to the current set of data in the ${entityTypeDisplay},
+     create a new version and mint a DOI for that version.`
+  }
+
+  const {
+    mutate: createOrUpdateDoi,
+    isPending: updateIsPending,
+    error: updateDoiError,
+  } = useCreateOrUpdateDOI({
+    onSuccess: () => {
+      displayToast('The DOI was successfully updated.', 'success')
+    },
+  })
+
+  const isLoading =
+    updateIsPending || isLoadingCurrentUser || isLoadingEntity || isLoadingDoi
+
+  const wasModifiedViaAPI = Boolean(doi && getWasModifiedViaAPI(doi))
+
+  useEffect(() => {
+    if (doi) {
+      setFormData(convertDoiToFormData(doi))
+    }
+    // If there is no DOI, then pre-fill the form.
+    if (currentUser && entity && doi === null) {
+      const newFormData = {
+        titles: [entity.name],
+        creators: new Array<string>(),
+        resourceTypeGeneral: getSuggestedResourceTypeGeneral(
+          convertToEntityType(entity.concreteType),
+        ),
+        publicationYear: new Date().getFullYear(),
+      } satisfies DoiFormData
+      if (currentUser.lastName && currentUser.firstName) {
+        newFormData.creators.push(
+          `${currentUser.lastName}, ${currentUser.firstName}`,
+        )
+      }
+      setFormData(newFormData)
+    }
+  }, [currentUser, doi, entity])
+
+  function onSave() {
+    const requestDoi: V2Doi = convertFormDataToDoi(formData)
+    requestDoi.objectType = objectType
+    requestDoi.objectId = objectId
+    requestDoi.objectVersion = selectedVersionNumber
+    requestDoi.etag = doi?.etag
+    createOrUpdateDoi({
+      doi: requestDoi,
+      concreteType: 'org.sagebionetworks.repo.model.doi.v2.DoiRequest',
+    })
+  }
+
+  const dialogContent = (
+    <>
+      <Typography variant={'body1'} gutterBottom>
+        Minting a DOI allows you to credit contributors and makes it easy to
+        cite your work.
+      </Typography>
+      <Typography variant={'body1'} gutterBottom>
+        Note that while the DOI and its associated metadata will be publicly
+        visible outside of Synapse, your data will still adhere to its existing
+        access conditions. If your data is private, it will remain restricted,
+        but the DOI will still be listed.
+        {/* Once a DOI is minted, the resource will be added to the Synapse Data Catalog. */}
```

Comment:
This was in the design, but is it true?

Reply:
@thomasyu888 ?  When a DOI is minted for an entity (let's assume the entity is made public), is there a process that will eventually discover and add a row to [the Data Catalog table](https://www.synapse.org/Synapse:syn61609402)?

Reply:
This is _not_ true right now, but I think that was Alberto's intent for this to be true